python-package/xgboost/spark/core.py

```diff
@@ -1092,12 +1093,86 @@ def _get_feature_col(
             )
         return features_col, feature_col_names
 
+    def _get_pred_contrib_col_name(self) -> Optional[str]:
+        """Return the pred_contrib_col col name"""
+        pred_contrib_col_name = None
+        if (
+            self.isDefined(self.pred_contrib_col)
+            and self.getOrDefault(self.pred_contrib_col) != ""
+        ):
+            pred_contrib_col_name = self.getOrDefault(self.pred_contrib_col)
+
+        return pred_contrib_col_name
+
+    def _out_schema(self) -> str:
+        """The returned type of the user-defined function. The value must
+        be a DDL-formatted type string."""
+
+        return (
+            f"{pred.prediction} double, {pred.pred_contrib} array<double>"
+            if self._get_pred_contrib_col_name() is not None
+            else "double"
+        )
+
+    def _get_predict_func(self) -> Callable:
+        """Return the true prediction function which will be running on the executor side"""
+
+        predict_params = self._gen_predict_params_dict()
+        pred_contrib_col_name = self._get_pred_contrib_col_name()
+
+        def _predict(
+            model: XGBModel, X: ArrayLike, base_margin: Optional[np.ndarray]
```

Comment:
`base_margin` is not necessarily `np.ndarray`. Let's stick with `ArrayLike`.

Reply:
Done